src/app/tests/TestSceneTable.cpp

```diff
@@ -0,0 +1,852 @@
+
+static app::Clusters::Scenes::Structs::AttributeValuePair::Type OOPairs[1];
+static app::Clusters::Scenes::Structs::AttributeValuePair::Type LCPairs[2];
+static app::Clusters::Scenes::Structs::AttributeValuePair::Type CCPairs[8];
+
+static uint8_t OO_buffer[scenes::kMaxFieldsPerCluster] = { 0 };
+static uint8_t LC_buffer[scenes::kMaxFieldsPerCluster] = { 0 };
+static uint8_t CC_buffer[scenes::kMaxFieldsPerCluster] = { 0 };
+
+/// @brief Simulates a Handler where Endpoint 1 supports onoff and level control and Endpoint 2 supports onoff and color control
+class TestSceneHandler : public scenes::DefaultSceneHandlerImpl
+{
+public:
+    TestSceneHandler() = default;
+    ~TestSceneHandler() override {}
+
+    // Fills in cluster buffer and adjusts its size to lower than the maximum number of cluster per scenes
+    virtual void GetSupportedClusters(EndpointId endpoint, Span<ClusterId> & clusterBuffer) override
+    {
+        ClusterId * buffer = clusterBuffer.data();
+        if (endpoint == TEST_ENDPOINT1)
+        {
+            if (clusterBuffer.size() >= 2)
+            {
+                buffer[0] = ON_OFF_CID;
+                buffer[1] = LV_CTR_CID;
+                clusterBuffer.reduce_size(2);
+            }
+        }
+        else if (endpoint == TEST_ENDPOINT2)
+        {
+            if (clusterBuffer.size() >= 2)
+            {
+                buffer[0] = ON_OFF_CID;
+                buffer[1] = CC_CTR_CID;
+                clusterBuffer.reduce_size(2);
+            }
+        }
+    }
+
+    // Default function only checks if endpoint and clusters are valid
+    bool SupportsCluster(EndpointId endpoint, ClusterId cluster) override
+    {
+        bool ret = false;
+        if (endpoint == TEST_ENDPOINT1)
+        {
+            if (cluster == ON_OFF_CID || cluster == LV_CTR_CID)
+            {
+                ret = true;
+            }
+        }
+
+        if (endpoint == TEST_ENDPOINT2)
+        {
+            if (cluster == ON_OFF_CID || cluster == CC_CTR_CID)
+            {
+                ret = true;
+            }
+        }
+
+        return ret;
+    }
+
+    /// @brief Simulates save from cluster, data is already in an EFS struct but this isn't mandatory
+    /// @param endpoint target endpoint
+    /// @param cluster  target cluster
+    /// @param serialisedBytes data to serialize into EFS
+    /// @return success if successfully serialized the data, CHIP_ERROR_INVALID_ARGUMENT if endpoint or cluster not supported
+    CHIP_ERROR SerializeSave(EndpointId endpoint, ClusterId cluster, MutableByteSpan & serialisedBytes) override
+    {
+        CHIP_ERROR err = CHIP_ERROR_INVALID_ARGUMENT;
+
+        if (endpoint == TEST_ENDPOINT1)
+        {
+            switch (cluster)
+            {
+            case ON_OFF_CID:
+                err = CHIP_NO_ERROR;
+                memcpy(serialisedBytes.data(), OO_buffer, scenes::kMaxFieldsPerCluster);
+                serialisedBytes.reduce_size(15); // Used memory for OnOff TLV
```

Comment:
How are you certain of the 15 and 27?

Reply:
By testing without reducing the size, I was able to confirm how much space was taken by the different clusters when everything is present: parameters id, tags, etc. It yielded the numbers currently used in the test.

Reply:
This is very risky as it is not robust to changes. You should keep track of the serialized size of the OO_buffer you reuse, so that you never have to put the size here explicitly.

Reply:
Changed to writter.GetWrittentLength() to obtain size.